plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/viewers/MapViewer.java

```diff
@@ -94,30 +99,30 @@
      * @param parent
      *      the parent control
      */
-    public MapViewer(Composite parent) {
-        this(parent, SWT.DOUBLE_BUFFERED | SWT.MULTI );
+    public MapViewer(Composite parent, MapPart mapPart) {
+        this(parent, mapPart, SWT.DOUBLE_BUFFERED | SWT.MULTI );
     }
 
     /**
-     * Creates a table viewer on a newly-created table control under the given
-     * parent. The table control is created using the given style bits. The
-     * viewer has no input, no content provider, a default label provider, no
-     * sorter, and no filters. The table has no columns.
+     * Creates a table viewer on a newly-created table control under the given parent. The table
+     * control is created using the given style bits. The viewer has no input, no content provider,
+     * a default label provider, no sorter, and no filters. The table has no columns.
      * 
-     * @param parent
-     *      the parent control
-     * @param style Use SWT.SINGLE or SWT.MULTI to control the use of tiles, 
-     *      SWT.DOUBLE_BUFFERED and SWT.NO_BACKGROUND can be used to configure the canvas. 
+     * @param parent the parent control
+     * @param mapPart the map part of this viewer.
+     * @param style Use SWT.SINGLE or SWT.MULTI to control the use of tiles, SWT.DOUBLE_BUFFERED and
+     *        SWT.NO_BACKGROUND can be used to configure the canvas.
      */
-    public MapViewer(Composite parent, int style) {
-        if( (style & SWT.MULTI) == SWT.MULTI ){
-            viewport = new ViewportPaneTiledSWT(parent, style, this );
-        }
-        else if( (style & SWT.SINGLE) == SWT.SINGLE){
-            viewport = new ViewportPaneSWT(parent, style, this );            
-        }
-        else {
-            viewport = new ViewportPaneSWT(parent, this );   
+    public MapViewer(Composite parent, MapPart mapPart, int style) {
+        Assert.isNotNull(mapPart, "MapPart required");
```

Comment:
`Objects.requireNonNull(mapPart, "MapPart required");` would be fine too. Since uDig haven't used both its fine to use Eclipse implementation.